framework/db/Connection.php

```diff
@@ -713,12 +713,22 @@ public function transaction(callable $callback, $isolationLevel = null)
             }
         } catch (\Exception $e) {
             if ($transaction->isActive && $transaction->level === $level) {
```

Comment:
Should it be a method in `Connection` with arguments `($transaction, $level)` or method in `Transaction` with arguments `($level)`. Also these two blocks will become one when minimum PHP version will be required 7.